library/manageiq_alert.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/python
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_alert
+description: The manageiq_alert module supports adding, updating and deleting alert definitions in ManageIQ.
+short_description: management of alerts in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  miq_verify_ssl:
+    description:
+      - whether SSL certificates should be verified for HTTPS requests
+    required: false
+    default: True
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+  description:
+    description:
+      - the alert definition description in manageiq
+    required: false
+    default: null
+  entity:
+    description:
+      - the entity to base the alert on in manageiq
+    required: false
+    choices: ['node', 'vm', 'host', 'storage', 'cluster', 'ems', 'miq_server', 'middleware_server']
+    default: null
```

Comment:
Please make sure doc says they're required except when deleting.  And sounds like `description` should always be required.